### PR TITLE
Turn more enum constants into constexpr

### PR DESCRIPTION
**Pull request by JosJuice**
https://bugs.dolphin-emu.org/issues/11692#note-7

### CODE REVIEW

**Comment by lioncash** [comment]
`static` can be omitted from the constants. constexpr (and const) internally link by default.

**Comment by JosJuice** [comment]
IntelliSense showed errors when I tried to use `constexpr` without `static` inside classes.

**Comment by lioncash** [comment]
You may also want to make flags or values intended to be unsigned as `u32` and not any type smaller if you want the signedness for them to be preserved in operations. e.g. `u8 | u8`, `u16 | u16` and `u8 | u16`, `u16 | u8` (ditto for other binary bitwise arithmetic) are cases that will result in the value being promoted to a signed `int`, not an `unsigned int` or whatever the source unsigned type was.

**Comment by lioncash** [comment]
Mainly referring to the cases of enums where there was simply:

```cpp
enum
{
  A,
  B,
  C,
  ...
};
```

without a type qualifier (defaulting to int), being given a size smaller than that.

**Comment by JosJuice** [comment]
There is some code in Dolphin that treats the constants as the sizes I set them to. For instance: https://github.com/dolphin-emu/dolphin/blob/360f2b4a2f5e2587a4e49952c885c9624f3c9a6a/Source/Core/Core/HW/WiimoteReal/IOWin.cpp#L452

And these arrays that are present for most extension controllers: https://github.com/dolphin-emu/dolphin/blob/360f2b4a2f5e2587a4e49952c885c9624f3c9a6a/Source/Core/Core/HW/WiimoteEmu/Extension/Classic.cpp#L25

The sizes I used in the WiimoteEmu code are also consistent with the sizes of constexpr constants that already exist there (e.g. Turntable.cpp), so I thought it would be best to do it that way.

**Comment by lioncash** [comment]
> The sizes I used in the WiimoteEmu code are also consistent with ...

Sounds good to me

File: Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h [hunk 1]
```diff
@@ -88,47 +88,43 @@ class Classic : public EncryptedExtension
...
+  static constexpr u16 BUTTON_Y = 0x2000;
+  static constexpr u16 BUTTON_A = 0x1000;
+  static constexpr u16 BUTTON_X = 0x0800;
+  static constexpr u16 BUTTON_ZR = 0x0400;
+  static constexpr u16 PAD_LEFT = 0x0200;
+  static constexpr u16 PAD_UP = 0x0100;
+
+  static constexpr u8 CAL_STICK_CENTER = 0x80;
+  static constexpr u8 CAL_STICK_RANGE = 0x7f;
+  static constexpr int CAL_STICK_BITS = 8;
```
Comment (BhaaLseN) [inline Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h]: Any particular reason you left those as (signed) int insted of `u32` for example?
Everything else is unsigned, and I can hardly think of a case where _a number of bits_ can be negative.

Comment (JosJuice) [reply Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h]: Just for consistency with existing constants in Turntable.cpp. Should I change it? To `u8`?

Comment (BhaaLseN) [reply Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h]: Probably doesn't change too much there; was mostly a question. What does everyone else think about this?

Comment (leoetlino) [reply Source/Core/Core/HW/WiimoteEmu/Extension/Classic.h]: I think int is fine here. It's consistent with other constants as JosJuice said and `std::numeric_limits<T>::digits` also uses int for numbers of bits.

I'd pick size_t if we do decide to change them all in the future, though.

**Comment by leoetlino** [comment]
This needs a rebase

**Pull request merged**